### ISSUE TITLE
[REQUEST] Real-time sensors for water flow, pressure, and temperature?

### ISSUE THREAD

**Issue by Hedda** [MizterB/homeassistant-phyn#16]
Can currently only see sensors for the average water flow-rate, water pressure, water temperature in this Phyn integration.

Is it possible to add real-time sensors as well for the water flow-rate, the water pressure, and the water temperature?

At least the water pressure and water temperature need to be real-time to be of any real use as those are key values to keep track of to prevent leaks or damage to appliances and frozen pipe bursts.

PS: The original Phyn app/software does show real-time data and for example warns of "High Base Pressure Alert" if the average water pressure is higher than 80 psi over the past 24-hour period or warns of "High-Pressure Warning" if the water pressure is much higher than normal for more than 1-minute and gives the recommendation to install a PRV (Pressure Reducing Valve), just as it gives a "Freeze Warning" or "Freeze Emergency" warnings depending if the temperature is too low or a combination of lower temperature and high pressure during a shorter period.

References:

https://phynsupport.my.salesforce-sites.com/FAQ/article_Datacategory1?id=3

https://helpcenter.phyn.com/help/i-got-a-high-base-pressure-alert-what-does-that-mean-and-what-should-i-do

https://helpcenter.phyn.com/help/what-are-the-normal-water-pressure-ranges-and-when-is-a-high-base-pressure-alert-sent

https://helpcenter.phyn.com/help/i-got-a-freeze-warning-alert-what-does-this-mean-and-what-should-i-do-c8c769b6

https://helpcenter.phyn.com/help/what-is-a-pressure-regulating-valve-prv

**Comment by bytenik** [comment]
I agree, the real-time sensors are critical for this device.

**Comment by MizterB** [comment]
Yes it is possible - @jordanruthe managed to reverse engineer the MQTT web sockets that contain this data, and well as enable additional devices.  I know it's not ready for prime-time yet, but you should keep an eye on his forks for progress on this.  Personally, I don't have time to take it further right now.

**Comment by Hedda** [comment]
> Yes it is possible - @jordanruthe managed to reverse engineer the MQTT web sockets that contain this data, and well as enable additional devices. I know it's not ready for prime-time yet, but you should keep an eye on his forks for progress on this.

Ah, noticed now that forks by @jordanruthe and @rsocko and @timbba80 and @opq8 all implemented different patches in their own aiophyn and homeassistant-phyn repositories :

https://github.com/jordanruthe/homeassistant-phyn

https://github.com/jordanruthe/aiophyn

https://github.com/rsocko/homeassistant-phyn

https://github.com/rsocko/aiophyn

https://github.com/timbba80/homeassistant-phyn

https://github.com/opq8/homeassistant-phyn

That is in addition to the upstream repositories by @MizterB

https://github.com/MizterB/homeassistant-phyn/

https://github.com/MizterB/aiophyn
